enhancements/sig-architecture/123-addon-multiple-configs/README.md

```diff
@@ -0,0 +1,230 @@
+
+### Goals
+
+- Enhance `ClusterManagementAddon` and `ManagementClusterAddon` API to allow user to configure mutiple addon configs with same GVK.
+
+### Non-Goals
+
+- `ClusterManagementAddon` `spec.supportedConfigs` does not allow user to configure mutiple addon configs with same GVK in the `defaultConfig`.
+
+## Proposal
+
+### User Stories
+
+1. The admin can configure the `ClusterManagementAddon` API to set mutilple configs with same gvk per install strategy.
+2. The admin can configure the `ManagementClusterAddon` API to set mutilple configs with same gvk for specific cluster.
+
+## Design Details
+
+### API changes
+
+`cma.spec.installStrategy.placements[].configs` and `mca.spec.configs` already support mutiple configs. 
+
+```go
+	// Configs is the configuration of managedClusterAddon during installation.
+	// User can override the configuration by updating the managedClusterAddon directly.
+	// +optional
+	Configs []AddOnConfig `json:"configs,omitempty"`
+```
+
+### Effective addon configs
+
+As the motivation mentioned, today there are 3 places to set addon configurations. 
+
+Currently, since each GVK supports only one config, the GVK is used as the identifier for configs. When configs with same GVK configured in multiple places, it follows the override sequence and 
+only one config is listed as the effective config in the mca.status.
+
+With the support for multiple configs with the same GVK, `GVK + namespace + name` becomes the identifier. 
+We need to reconsider the logic for setting mca.status when configuring multiple configs in multiple places.
+
+Using the example below (omitting the group and namespace for simplicity), we present three options. 
+We do not expect to support all, but want to involve discussion and finally only choose one.
+
+```yaml
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ClusterManagementAddOn
+metadata:
+  name: example
+spec:
+  supportedConfigs:
+    - resource: r1
+      defaultConfig:
+        name: n1
+    - resource: r2
+  installStrategy:
+    type: Placements
+    placements:
+      - name: example
+        namespace: example
+        configs:
+          - resource: r2
+            name: n1
+          - resource: r2
+            name: n2
+
+
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ManagedClusterAddOn
+metadata:
+  name: example
+  namespace: cluster1
+spec:
+  installNamespace: open-cluster-management-agent-addon
+  configs:
+    - resource: r2
+      name: n3
+```
+
+#### Merge
+
+With `GVK + namespace + name` as the identifier for each config, there is no override concept now actually, all the configured configs will be listed in the mca.status. 
+
+Configs will be ordered, with the last configured on top. For example, configs in mca will be on top of cma configs, and mca configs with higher indices 
+will be on top of those with lower indices.
+
+The addon's own controller will determine how to use the configs (e.g., choose one, read all, etc).
```

Comment:
In the merge option, the configs in mca and cma will be listed in order. For example, if c1 is an `OpenTelemetryCollector` cr configured to collect traces and c2 is to collect logs, c3 is a new config to collect logs, the controller will see c3, c2, c1, it could always choose the first one to collect logs which is c3, and could also choose the other one for traces which is c1. Does this make sense from addon's view?

Reply:
I understand the configs will be shown in order, but this is insufficient to inform the addon if a resource comes from the global config in CMA or a cluster-specific config in MCA. For example, assume a OTELCollector config that would collect logs (c1), but then for a given cluster we are troubleshooting something and we want to send only metrics & traces so we specify (c2 & c3, two separate `OpenTelemetryCollector`). From the addon perspective, it's not clear what config should be used even if they are sorted by (c3, c2, c1). The only alternative is to go through the resources in the Spec of MCA and calculate the intersection with the Status.

After discussing with the team from our POV we prefer option 2. However, if the OCM team favours option 1 then we also do not oppose since we have the workaround to calculate the intersection.

Reply:
OK, this case makes sense. For this proposal, the option 2 can be selected to be implemented first, since it makes more sense for real use cases. As for the "global configs" issues, to avoid override, the workaround would be to configure the "global configs" at both cma and cma. 

Will update the doc and leave the other 2 options for future needs.